plugins/source/azure/client/services.go

```diff
@@ -0,0 +1,130 @@
+	if err != nil {
+		return services, err
+	}
+	services.ArmbatchAccounts = ArmbatchAccounts
+
+	ArmcdnProfiles, err := armcdn.NewProfilesClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmcdnProfiles = ArmcdnProfiles
+
+	ArmcdnEndpoints, err := armcdn.NewEndpointsClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmcdnEndpoints = ArmcdnEndpoints
+
+	ArmcomputeDisks, err := armcompute.NewDisksClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmcomputeDisks = ArmcomputeDisks
+
+	ArmcomputeVirtualMachines, err := armcompute.NewVirtualMachinesClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmcomputeVirtualMachines = ArmcomputeVirtualMachines
+
+	ArmcomputeVirtualMachineScaleSets, err := armcompute.NewVirtualMachineScaleSetsClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmcomputeVirtualMachineScaleSets = ArmcomputeVirtualMachineScaleSets
+
+	ArmcontainerregistryRegistries, err := armcontainerregistry.NewRegistriesClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmcontainerregistryRegistries = ArmcontainerregistryRegistries
+
+	ArmcontainerserviceManagedClusters, err := armcontainerservice.NewManagedClustersClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmcontainerserviceManagedClusters = ArmcontainerserviceManagedClusters
+
+	ArmcosmosAccounts, err := armcosmos.NewDatabaseAccountsClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmcosmosAccounts = ArmcosmosAccounts
+
+	ArmdatalakestoreAccounts, err := armdatalakestore.NewAccountsClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmdatalakestoreAccounts = ArmdatalakestoreAccounts
+
+	ArmdatalakeanalyticsAccounts, err := armdatalakeanalytics.NewAccountsClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmdatalakeanalyticsAccounts = ArmdatalakeanalyticsAccounts
+
+	ArmeventhubNamespaces, err := armeventhub.NewNamespacesClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmeventhubNamespaces = ArmeventhubNamespaces
+
+	ArmeventhubNetworkRuleSets, err := armeventhub.NewNamespacesClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmeventhubNetworkRuleSets = ArmeventhubNetworkRuleSets
+
+	ArmfrontdoorDoors, err := armfrontdoor.NewFrontDoorsClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmfrontdoorDoors = ArmfrontdoorDoors
+
+	ArmkeyvaultVaults, err := armkeyvault.NewVaultsClient(subscriptionId, azCred, nil)
+	if err != nil {
+		return services, err
+	}
+	services.ArmkeyvaultVaults = ArmkeyvaultVaults
```

Comment:
a good idea to init in-place.
I'd go even further and do it like this:
```go
var err error
var services Services

services.ClientA, err = initClientA(...)
if err != nil {
    return services, err
}
...
```

Maybe I'll add it to my PR codegen as well